### ISSUE TITLE
Pyne having issues finding HDF5 if not using system HDF5

### ISSUE THREAD

**Issue by ljacobson64** [pyne/pyne#928]
It used to be the case that pyne would automatically find the HDF5 in your `LD_LIBRARY_PATH`, but that appears to have changed in #908. Now it gives an error:

```
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
CMake Error at /groupspace/cnerg/users/jacobson/opt/precompiled/cmake/3.9.4/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS
  HDF5_HL_LIBRARIES)
Call Stack (most recent call first):
  /groupspace/cnerg/users/jacobson/opt/precompiled/cmake/3.9.4/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  /groupspace/cnerg/users/jacobson/opt/precompiled/cmake/3.9.4/share/cmake-3.9/Modules/FindHDF5.cmake:839 (find_package_handle_standard_args)
  CMakeLists.txt:45 (FIND_PACKAGE)

-- Configuring incomplete, errors occurred!
```

If I add the `-DHDF5_INCLUDE_DIRS=...` definition to my build command, then the error changes:

```
CMake Error at /groupspace/cnerg/users/jacobson/opt/precompiled/cmake/3.9.4/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS
  HDF5_HL_LIBRARIES) (found version "1.8.13")
Call Stack (most recent call first):
  /groupspace/cnerg/users/jacobson/opt/precompiled/cmake/3.9.4/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  /groupspace/cnerg/users/jacobson/opt/precompiled/cmake/3.9.4/share/cmake-3.9/Modules/FindHDF5.cmake:839 (find_package_handle_standard_args)
  CMakeLists.txt:45 (FIND_PACKAGE)

-- Configuring incomplete, errors occurred!
```

I tried setting the `HDF5_ROOT` environment variable and setting various other build definitions, but none of them worked.

**Comment by ljacobson64** [comment]
I'd also like to point out that that same PR drastically increases the time it takes to build pyne, mainly because `cram.c` has 169,136 lines.

**Comment by scopatz** [comment]
Yeah, @asmeurer and I are still working through some of the cram and bateman issues.  Sorry about that. We are working on a fix.

**Comment by scopatz** [comment]
As per the HDF5 thing, have you tried using the `setup.py` variables `--hdf5` or `--deps-root`.  Specifically, deps-root is the root path for the dependencies and should work.  This is set by default to the root path of where the compiler is found, and this is the first place where the dependencies are looked for.  

Also, I am not sure that we should be using `$LD_LIBRARY_PATH` at build time, since that is specifically for dynamic linking at run time.  I think that `$LIBRARY_PATH` is the right envvar to rely on, but that is probably being overshadowed by deps-root too.

**Comment by ljacobson64** [comment]
I have tried `--hdf5` and `--moab` but neither seemed to work. I have not tried `--deps-root` but I'm not sure what I would even set it to since HDF5 and MOAB are in different directories.

With PR #930, I'm able to build using `--hdf5` and `--moab` with an empty `$LD_LIBRARY_PATH`. I feel this is the easiest way to tell users to specify the locations of HDF5 and MOAB since it does not require them to set any environment variables and it does not require them to set things like `-DMOAB_LIBRARY` and `-DMOAB_INCLUDE_DIR`.

It does look in the `$LD_LIBRARY_PATH` to try to find MOAB, but only if `-DMOAB_ROOT` isn't set. (`--moab` sets `-DMOAB_ROOT`.)

**Comment by scopatz** [comment]
The problem with #930 is that it breaks building off of a conda environment and reverts to those users having to set `$LD_LIBRARY_PATH` just to import.  I think the right thing to do here is to have `--hdf5` and `--moab` give appropriate hints that take precedence dep-roots, rather than reverting the changes.  I hope that this makes sense.  I'm going to try to put a PR in right now to this effect.

**Comment by ljacobson64** [comment]
Fixed by #932.
